src/pages/signin/ResendValidationForm.js

```diff
@@ -88,7 +88,7 @@ class ResendValidationForm extends React.Component {
 
     render() {
         const isNewAccount = !this.props.account.accountExists;
-        const isOldUnvalidatedAccount = this.props.account.accountExists && !this.props.account.validated;
+        const isOldUnvalidatedAccount = this.props.account.accountExists && !this.props.account.validated && !this.props.account.validateCodeExpired;
```

Comment:
I'll try to explain this again in another way...

- An "old unvalidated account" would be one that "exists" and is "not validated".
- Having an expired validate code does not tell us anything about whether it is an "old unvalidated account"
- It only tells us that there is an expired validate code.

Therefore we should remove the last check like so...

```suggestion
        const isOldUnvalidatedAccount = this.props.account.accountExists && !this.props.account.validated;
```